### PR TITLE
Add support for mode icon overrides by route

### PR DESCRIPTION
**Pull request by amy-corson-ibigroup**
**Description:**
Makes mode icon overrides configurable! Changes mode icon displayed based on config without changing anything else.

**PR Checklist:**
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [ ] Are appropriate Typescript types implemented?

### CODE REVIEW

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
I'm not a huge fan of all this new code... Could we instead just change each `<ModeIcon ...` to include a new routeId prop? And then inside the modeIcon component we can do all the logic we need. That way we don't have to write all this new code and don't have to add any new features to config.yaml?

**Comment by amy-corson-ibigroup** [comment]
So would you add all the key value pairs to the custom mode icon renderer in this config.js file? That seems like it would get messy? 

I thought of adding a prop to the `ModeIcon` but the problem I ran into is the `LegIconWithAlly` component we pass to OTP-UI. I guess we could add the prop in OTP-UI too but keeping it in otp-rr seemed a little easier?

**Comment by miles-grant-ibigroup** [comment]
> So would you add all the key value pairs to the custom mode icon renderer in this config.js file? That seems like it would get messy?
> 
It's messier than yours that's true but I think I prefer a cleaner otp-rr to a cleaner config. Happy to punt this to the tech call though if you think we should get everyone's feedback

> I thought of adding a prop to the `ModeIcon` but the problem I ran into is the `LegIconWithAlly` component we pass to OTP-UI. I guess we could add the prop in OTP-UI too but keeping it in otp-rr seemed a little easier?

Oh grr... I still think it's cleaner than this though! An entire new set of methods for a feature like this is giving me the ick

File: lib/components/narrative/metro/route-block.tsx [hunk 1]
```diff
@@ -142,13 +149,15 @@ const RouteBlock = ({
...
+  console.log(leg)
+
   return (
     <>
       {showDivider && previousLegMode && <Divider>•</Divider>}
       <Wrapper className="route-block-wrapper">
         {leg.mode !== previousLegMode && (
           <LegIconWrapper>
-            <LegIcon height={28} leg={leg} />
+            <LegIcon height={28} leg={leg} routeId={leg.routeId || null} />
```
Comment (miles-grant-ibigroup) [inline lib/components/narrative/metro/route-block.tsx:160]: Why are we explicitly adding `leg.routeId` when we're already sending the entire `leg`?

File: lib/components/narrative/metro/route-block.tsx [hunk 2]
```diff
@@ -142,13 +149,15 @@ const RouteBlock = ({
     }
   }
 
+  console.log(leg)
```
Comment (miles-grant-ibigroup) [inline lib/components/narrative/metro/route-block.tsx:152]: oops!

File: lib/components/viewers/RouteRow.js [hunk 3]
```diff
@@ -192,6 +192,7 @@ export class RouteRow extends PureComponent {
                   }}
                   mode={getModeFromRoute(route)}
                   role="img"
+                  routeId={route.id}
```
Comment (miles-grant-ibigroup) [inline lib/components/viewers/RouteRow.js:195]: Could we just add this to the `leg` like we do with the long and short names?

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Some architecture questions

File: lib/components/narrative/metro/route-block.tsx (hunk 2 above)
Comment (amy-corson-ibigroup) [reply lib/components/narrative/metro/route-block.tsx]: you dont want to keep ?

File: lib/components/narrative/metro/route-block.tsx (hunk 1 above)
Comment (amy-corson-ibigroup) [reply lib/components/narrative/metro/route-block.tsx]: Yes good point!!

**Comment by amy-corson-ibigroup** [comment]
The leg prop was not making it through `ClassicLegIcon`, so this fix now also relies on https://github.com/opentripplanner/otp-ui/pull/707

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
A few more cleanup possibilities!

File: lib/components/narrative/line-itin/LegIconWithA11y.tsx [hunk 4]
```diff
@@ -19,7 +19,7 @@ const LegIconWithA11y = (props: any) => {
   const ariaLabel = isTransit(mode) ? getFormattedMode(mode, intl) : null
   return (
     <>
-      <LegIcon {...props} />
+      <LegIcon {...props} leg={leg} />
```
[the remark this answers is not in the document]
Comment (amy-corson-ibigroup) [reply lib/components/narrative/line-itin/LegIconWithA11y.tsx]: Good catch thank you!

File: lib/components/narrative/line-itin/LegIconWithA11y.tsx [hunk 5]
```diff
@@ -17,6 +17,7 @@ const LegIconWithA11y = (props: any) => {
   const { leg } = props
   const { mode } = leg
   const ariaLabel = isTransit(mode) ? getFormattedMode(mode, intl) : null
+  console.log(props)
```
Comment (miles-grant-ibigroup) [inline lib/components/narrative/line-itin/LegIconWithA11y.tsx:20]: Whoops!

**Review by miles-grant-ibigroup — APPROVED** [review]
One last clean up and then I think we're finally good to go! What a tiny PR this wound up being haha

File: lib/components/narrative/line-itin/LegIconWithA11y.tsx (hunk 5 above)
Comment (amy-corson-ibigroup) [reply lib/components/narrative/line-itin/LegIconWithA11y.tsx:20]: what is going on with me!!!

**Review by binh-dam-ibigroup — APPROVED** [review]

**Pull request merged**